src/Simplex/Chat/Store/Shared.hs

```diff
@@ -401,3 +403,17 @@ createWithRandomBytes' size gVar create = tryCreate 3
 
 encodedRandomBytes :: TVar ChaChaDRG -> Int -> IO ByteString
 encodedRandomBytes gVar n = atomically $ B64.encode <$> C.randomBytes n gVar
+
+checkContactIsUser :: DB.Connection -> User -> Contact -> IO Bool
+checkContactIsUser db User {userContactId} Contact {contactId} = do
+  isUser_ <-
+    maybeFirstRow fromOnly $
+      DB.query db "SELECT is_user FROM contacts WHERE contact_id = ?" (Only contactId)
+  pure $ fromMaybe False isUser_ || contactId == userContactId
+
+checkLDNIsUser :: DB.Connection -> User -> ContactName -> IO Bool
+checkLDNIsUser db User {userId} ldn = do
```

Comment:
combine to one function

Reply:
also this query does not account for the possibility of another corruption, where is_user is inconsistent with contact_id in user record

Reply:
SELECT 1 FROM users
  WHERE user_id = ?
    AND (local_display_name = ? || contact_id = ?)
LIMIT 1